src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java

```diff
@@ -660,6 +660,41 @@ public RemoteOutputsStrategyConverter() {
               + "cache misses and retries.")
   public boolean remoteDiscardMerkleTrees;
 
+  @Option(
+       name = "experimental_circuit_breaker_strategy",
+       documentationCategory = OptionDocumentationCategory.REMOTE,
+       defaultValue = "null",
+       effectTags = {OptionEffectTag.EXECUTION},
+       converter = CircuitBreakerStrategy.Converter.class,
+       help =
+            "Specifies the strategy for the circuit breaker to use. Available strategies are \"failure\". "
+                + "On invalid value for the option the behavior same as the option is not set.")
+  public CircuitBreakerStrategy circuitBreakerStrategy;
+
+  @Option(
+          name = "experimental_remote_failure_threshold",
+          defaultValue = "100",
+          documentationCategory = OptionDocumentationCategory.REMOTE,
+          effectTags = {OptionEffectTag.EXECUTION},
+          help =
+              "Sets the allowed number of failures in a specific time window after which it stops calling to the "
+                  + "remote cache/executor. By default the value is 100. Setting this to 0 or negative means "
+                  + "no limitation.")
+  public int remoteFailureThreshold;
+
+  @Option(
+          name = "experimental_remote_failure_window_size",
+          defaultValue = "60s",
+          documentationCategory = OptionDocumentationCategory.REMOTE,
+          effectTags = {OptionEffectTag.EXECUTION},
+          converter = RemoteDurationConverter.class,
+          help =
+              "The sliding window in which the failure count of the remote requests are computed. On zero or negative "
+                 + "value the failure duration is computed the whole duration of the execution."
+                 + "Following units can be used: Days (d), hours (h), minutes (m), seconds (s), and milliseconds (ms). "
+                 + "If the unit is omitted, the value is interpreted as seconds.")
+  public Duration remoteFailureWindowSize;
```

Comment:
nit: not sure if you can point to prior art on this, but WindowSize might be better specified as WindowInterval - size here connotes that there will be some counted number of requests, not a sliding temporal inclusion.